tests/suite/tracing_test.go

```diff
@@ -0,0 +1,270 @@
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+// This test can be flaky when waiting to see traces show up in the collector logs.
+var _ = Describe("Tracing", FlakeAttempts(2), Label("functional"), func() {
+	files := []string{
+		"hello-world/apps.yaml",
+		"hello-world/gateway.yaml",
+		"hello-world/routes.yaml",
+	}
+	var ns core.Namespace
+
+	var collectorPodName, helloURL, worldURL, helloworldURL string
+
+	BeforeEach(func() {
+		ns = core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "helloworld",
+			},
+		}
+
+		output, err := installCollector()
+		Expect(err).ToNot(HaveOccurred(), string(output))
+
+		collectorPodNames, err := resourceManager.GetPodNames(
+			collectorNamespace,
+			client.MatchingLabels{
+				"app.kubernetes.io/name": "opentelemetry-collector",
+			},
+		)
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(collectorPodNames).To(HaveLen(1))
+
+		collectorPodName = collectorPodNames[0]
+
+		Expect(resourceManager.Apply([]client.Object{&ns})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(files, ns.Name)).To(Succeed())
+		Expect(resourceManager.WaitForAppsToBeReady(ns.Name)).To(Succeed())
+
+		url := "http://foo.example.com"
+		helloURL = url + "/hello"
+		worldURL = url + "/world"
+		helloworldURL = url + "/helloworld"
+		if portFwdPort != 0 {
+			helloURL = fmt.Sprintf("%s:%s/hello", url, strconv.Itoa(portFwdPort))
+			worldURL = fmt.Sprintf("%s:%s/world", url, strconv.Itoa(portFwdPort))
+			helloworldURL = fmt.Sprintf("%s:%s/helloworld", url, strconv.Itoa(portFwdPort))
+		}
+	})
+
+	AfterEach(func() {
+		output, err := uninstallCollector()
+		Expect(err).ToNot(HaveOccurred(), string(output))
+
+		Expect(resourceManager.DeleteFromFiles(files, ns.Name)).To(Succeed())
+		Expect(resourceManager.DeleteNamespace(ns.Name)).To(Succeed())
+	})
+
+	updateGatewayClass := func() error {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.CreateTimeout)
+		defer cancel()
+
+		key := types.NamespacedName{Name: "nginx"}
+		var gwClass gatewayv1.GatewayClass
+		if err := k8sClient.Get(ctx, key, &gwClass); err != nil {
+			return err
+		}
+
+		gwClass.Spec.ParametersRef = &gatewayv1.ParametersReference{
+			Group: ngfAPI.GroupName,
+			Kind:  gatewayv1.Kind("NginxProxy"),
+			Name:  "nginx-proxy",
+		}
+
+		return k8sClient.Update(ctx, &gwClass)
+	}
+
+	sendTraceRequests := func(ctx context.Context, url string, count int) {
```

Comment:
```suggestion
	sendRequests := func(ctx context.Context, url string, count int) {
```

the name `sendTraceRequests` was confusing to me because I thought the function was checking tracing.